tests/async/test_browsertype_connect.py

```diff
@@ -241,3 +243,54 @@ async def handle_request(route: Route) -> None:
     assert await response.json() == {"foo": "bar"}
 
     remote.kill()
+
+@pytest.mark.only_browser("chromium")
+async def test_should_upload_large_file(
+    browser_type: BrowserType,
+    launch_server: Callable[[], RemoteServer],
+    playwright: Playwright,
+    server: Server,
+    tmp_path):
+    remote = launch_server()
+
+    browser = await browser_type.connect(remote.ws_endpoint)
+    context = await browser.new_context()
+    page = await context.new_page()
+
+    await page.goto(server.PREFIX + "/input/fileupload.html")
+    large_file_path = tmp_path / "200MB.zip"
+    data = b'A' * 1024
+    with large_file_path.open("wb") as f:
+        for i in range(0, 200 * 1024 * 1024, len(data)):
+            f.write(data)
+    input = page.locator('input[type="file"]')
+    events = await input.evaluate_handle(
+        """
+        e => {
+            const events = [];
+            e.addEventListener('input', () => events.push('input'));
+            e.addEventListener('change', () => events.push('change'));
+            return events;
+        }
+    """
+    )
+
+    await input.set_input_files(large_file_path)
+    assert await input.evaluate("e => e.files[0].name") == "200MB.zip"
+    assert await events.evaluate("e => e") == ["input", "change"]
+
+    file_upload = asyncio.Future()
+    def handler(request):
+        file_upload.set_result(request)
+        request.finish()
+    server.set_route("/upload", handler)
+
+    await page.click("input[type=submit]")
+    request = await file_upload
+    contents = request.args[b'file1'][0]
+    assert len(contents) == 200 * 1024 * 1024
+    assert contents[:1024] == data
+    assert contents[len(contents)-1024:] == data
+    match = re.search(rb'^.*Content-Disposition: form-data; name="(?P<name>.*)"; filename="(?P<filename>.*)".*$', request.post_body, re.MULTILINE)
```

Comment:
Let's consider using a proper multipart form parsing lib. This works for now, though!

Reply:
twisted can't get the file name out of the box yeah, you could get the name and it's value via `len(request.args['file1'][0])` but it seems like your approach is fine for now.